rudder-templates/src/main/scala/com/normation/templates/FillTemplatesService.scala

```diff
@@ -72,8 +72,32 @@ final case class STVariable(
   , isSystem  :Boolean
 )
 
+
 class FillTemplatesService extends Loggable {
 
+  /*
+   * The cache is managed template by template
+   * If a content is not in the cache, it will create the StringTemplate instance, and return it
+   *
+   */
+  private[this] var cache = Map[String,StringTemplate]()
+
+
+  def clearCache(): Unit = this.synchronized {
+    cache = Map()
+  }
+
+  def getTemplateFromContent(content: String): Box[StringTemplate] = this.synchronized {
+    cache.get(content) match {
+      case Some(template) => Full(template.getInstanceOf())
+      case _ =>
+        val template = new StringTemplate(content, classOf[NormationAmpersandTemplateLexer])
```

Comment:
I believe that can throw exception, no ? If so, can you wrapp it in `try/catch ex => Failure...`. And the same for `getInstanceOf` if needed.

Reply:
you are totally right